### PR TITLE
AARA-505, 506: TRACKING:  Implement RFA 02 card

### PR DESCRIPTION
**Pull request by anuroop15**
#### Which User story does this relate to (link)?
- https://osi-cwds.atlassian.net/browse/AARA-505
 https://osi-cwds.atlassian.net/browse/AARA-506

#### Brief feature description
-  Implemented RFA 02 edit and save mode pages with conditional logic. Also, couples API's from rails with fron-end view.

#### Does this PR block/effect other stories?
- NA

#### Do any commands needs to be run in order to run the project with this PR?
- NA

#### Code coverage percent diff as a result of this PR?
- 89.18% Statements 
 81.68% Branches 
 86.87% Functions 
89.36% Lines

### CODE REVIEW

**Comment by anuroop15** [comment]
jenkins retest this please

**Comment by anuroop15** [comment]
@lauherkcambria  RFA 02 should be defaulted to Edit mode as per design. I even spoke with Robert  and he confirmed the same. 

@nasar-cambria updated the date field to show Date on save.

**Comment by anuroop15** [comment]
jenkins retest this please

**Comment by lauherkcambria** [comment]
@anuroop15  `RFA 02 should be defaulted to Edit mode as per design. I even spoke with Robert and he confirmed the same.`

hahaha nevermind then! 😄  that being said though, should tracking then default to edit mode instead of show mode? I'll ask robert and put in a jira story if that's the case. 

Thanks for asking design and confirming this one. 👍

File: app/javascript/rfa_forms/rfa02_edit_view/backGroundCheckRows/liveScanRow.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        return (
+          <tr key={id + 'checkboxShow' + +index}>
+            <td id={id + 'checkboxShow' + index}>
+              <BinarySelectorField
+                type='checkbox'
+                key={index}
+                labelId={'checkLabel' + index}
+                label={docs.get('title')}
+                gridClassName='col-xs-12'
+                defaultChecked={docs.get('checked')} />
```
[the remark this answers is not in the document]
Comment (anuroop15) [reply app/javascript/rfa_forms/rfa02_edit_view/backGroundCheckRows/liveScanRow.js]: didn't get you. why would we use checkbox fields to be replaced with YesNoFieldShow??

File: app/javascript/rfa_forms/rfa02_edit_view/checkListTable.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+CheckListTable.defaultProps = {
+  colHeaders: [],
+  CardHeader: '',
+  editMode: false,
+  rowsComponent: null
+}
+
+export default CheckListTable
```
[the remark this answers is not in the document]
Comment (anuroop15) [reply app/javascript/rfa_forms/rfa02_edit_view/checkListTable.js]: good catch. 👍

**Comment by anuroop15** [comment]
@lauherkcambria addresses comments and fixed dateField issues on exemptions card.

File: app/javascript/rfa_forms/rfa02_edit_view/backGroundCheckRows/liveScanRow.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lauherkcambria) [reply app/javascript/rfa_forms/rfa02_edit_view/backGroundCheckRows/liveScanRow.js]: my misunderstanding

**Comment by anuroop15** [comment]
jenkins retest this please

**Pull request merged**